test-framework/common/src/main/java/io/quarkus/test/common/TestInstantiator.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.test.common;
+
+import java.lang.reflect.Constructor;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.CDI;
+
+public class TestInstantiator {
+
+    public static Object instantiateTest(Class<?> testClass) {
+
+        try {
+            BeanManager bm = CDI.current().getBeanManager();
+            Set<Bean<?>> beans = bm.getBeans(testClass);
+            Set<Bean<?>> nonSubClasses = new HashSet<>();
```

Comment:
What's this for? `SubstrateTest` is not a bean defining annotation so the subclasses should not be recognized as beans.

Reply:
Because you can have tests that extend other tests. It's not a common use case but we might as well make it work correctly. I realised this with the InjectedSpringBeansResourceIT problem (where the IT was not a SubstrateTest) but there are valid use cases for this.

Reply:
Ok. Makes sense.